### DISCUSSION TITLE
Spotify* alarm like Google Clock? (*or other audio apps)

### DISCUSSION THREAD

**Discussion by e-t-l** [BlackyHawky/Clock#123] [Ideas]
Is there interest in creating an option for the alarm to play audio from an app such as Spotify, Pandora, etc like the Google Clock can? 

This could be easily accomplished natively using Intents. In its most basic form, the alarm would trigger an arbitrary intent, which the user would be responsible for crafting in order to activate their desired audio app. Ideally, a more sophisticated implementation would have a template for each app (perhaps something users could contribute) so that the user just make a selection in the GUI for the desired app and audio track/playlist. A middle-ground implementation might be to use the [Tasker Plugin Library](https://tasker.joaoapps.com/pluginslibrary.html) to allow Alarms to trigger Tasker commands (or in similar FOSS apps like Automation) which could then trigger an audio app.

Big apps like [Spotify](https://spotify.github.io/android-sdk/) and [Pandora](https://developer.pandora.com/docs/key-concepts/apis/) have their own APIs for allowing other apps to trigger their audio, but using any of the above methods, we could make it compatible with a much wider array of audio apps, like Newpipe and podcast apps (whereas Google Clock is limited to just Spotify, Pandora, YT Music, and Calm).

**Comment by BlackyHawky** [comment]
I don't plan on working on this and I'm not sure I want it for this app.
